config/vhosts/default.conf

```diff
@@ -1,8 +1,108 @@
+                                      "application/wasm" \
+                                      "application/x-font-ttf" \
+                                      "application/x-javascript" \
+                                      "application/x-web-app-manifest+json" \
+                                      "application/xhtml+xml" \
+                                      "application/xml" \
+                                      "font/eot" \
+                                      "font/opentype" \
+                                      "font/otf" \
+                                      "font/ttf" \
+                                      "image/bmp" \
+                                      "image/svg+xml" \
+                                      "image/vnd.microsoft.icon" \
+                                      "text/cache-manifest" \
+                                      "text/calendar" \
+                                      "text/css" \
+                                      "text/html" \
+                                      "text/javascript" \
+                                      "text/plain" \
+                                      "text/markdown" \
+                                      "text/vcard" \
+                                      "text/vnd.rim.location.xloc" \
+                                      "text/vtt" \
+                                      "text/x-component" \
+                                      "text/x-cross-domain-policy" \
+                                      "text/xml"
+
+    </IfModule>
+
+    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+    # Map the following filename extensions to the specified
+    # encoding type in order to make Apache serve the file types
+    # with the appropriate `Content-Encoding` response header
+    # (do note that this will NOT make Apache compress them!).
+    #
+    # If these files types would be served without an appropriate
+    # `Content-Encoding` response header, client applications (e.g.:
+    # browsers) wouldn't know that they first need to uncompress
+    # the response, and thus, wouldn't be able to understand the
+    # content.
+    #
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
+    # https://httpd.apache.org/docs/current/mod/mod_mime.html#addencoding
+
+    <IfModule mod_mime.c>
+        AddEncoding gzip              svgz
+    </IfModule>
+
+</IfModule>
+
 <VirtualHost *:80>
     ServerAdmin webmaster@localhost
     DocumentRoot "/var/www/html"
     ServerName localhost
 	<Directory "/var/www/html/">
 		AllowOverride all
 	</Directory>
-</VirtualHost>
\ No newline at end of file
+</VirtualHost>
+<VirtualHost *:443>
+    ServerAdmin webmaster@localhost
+    ServerAlias localhost
+    DocumentRoot "/var/www/html"
+    ServerName localhost
+	<Directory "/var/www/html/">
+		AllowOverride all
+	</Directory>
+
+    SSLEngine on
```

Comment:
I'm a little bit afraid of self signed certificates prebuild, because a lot of people use this stack as it is on public systems and this could introduce a big security hole. Maybe we can generate them during the webserver build somehow. Or we create a how to in the readme and comment the apache config here out, so apache is able to start for example.

Reply:
Ok. Should I also remove the self signed certificates from the commit?

Reply:
We could do it this way I think: 

* add a ssl volume unter `/config/ssl` ( there we can put ssl certs, and the user can switch them easily ) 
* During the the container build process we generate a fresh self signed ssl each time the container gets build. 
- generate CSR in the /config/ssl dir: 
```
openssl req -new -newkey rsa:2048 -nodes \
    -keyout www.example.com.key -out www.example.com.csr \
    -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
```
- Generate the cert 
```
openssl req -new -newkey rsa:2048 -days 365 -nodes -x509 \
    -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com" \
    -keyout www.example.com.key  -out www.example.com.cert
```

Please make sure that the keys pop up in the ssl directory. 

That is a way i could imagine could work. What do you think ?

Reply:
This seems fine. Sorry for the delayed response.

Reply:
Please take a look at a newly pushed commit with subject `Add certificate generation`. This will generate the certificates build time and also populates them into `config/ssl` directory.